From de251bbd58568f6f04fd05af4fc1aaecbcbab04a Mon Sep 17 00:00:00 2001
From: "Daniel F. Dickinson" <cshored@thecshore.com>
Date: Mon, 6 Feb 2023 11:07:58 +0300
Subject: [PATCH 1/3] uhttpd: Fix CGI finds wrong binaries due to inconsistent
 PATH

The PATH of the uhttpd CGI environment does not match the profile or init PATH
which results in luci finding the wrong binaries compared to the command line or
initscripts, when there is more than one binary of the same name.
---
 main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.c b/main.c
index de5f2c3..dda787b 100644
--- a/main.c
+++ b/main.c
@@ -190,7 +190,7 @@ static void init_defaults_pre(void)
 	conf.max_connections = 100;
 	conf.realm = "Protected Area";
 	conf.cgi_prefix = "/cgi-bin";
-	conf.cgi_path = "/sbin:/usr/sbin:/bin:/usr/bin";
+	conf.cgi_path = "/usr/sbin:/usr/bin:/sbin:/bin";
 	INIT_LIST_HEAD(&conf.cgi_alias);
 	INIT_LIST_HEAD(&conf.lua_prefix);
 #if HAVE_UCODE
-- 
2.34.1.windows.1

